torch/csrc/api/src/data/init.cpp

```diff
@@ -0,0 +1,479 @@
+      .def(
+          "load",
+          &DistributedRandomSampler::load,
+          "Deserializes the `Sampler` from the `archive`",
+          py::arg("archive"))
+      .def(
+          "index",
+          &DistributedRandomSampler::index,
+          "Returns the current index of the `DistributedRandomSampler`");
+
+  /// DistributedSequentialSampler
+  py::class_<DistributedSequentialSampler>(
+      data, "DistributedSequentialSampler", distributed_sampler)
+      .def(
+          py::init<size_t, size_t, size_t, bool>(),
+          "Create and return a new `DistributedSequentialSampler` instance",
+          py::arg("size"),
+          py::arg("num_replicas") = 1,
+          py::arg("rank") = 0,
+          py::arg("allow_duplicates") = true)
+      .def(
+          "reset",
+          &DistributedSequentialSampler::reset,
+          "Resets the `Sampler`'s internal state. Typically called before a new epoch. Optionally, accepts a new size when reseting the sampler",
+          py::arg("new_size"))
+      .def(
+          "next",
+          &DistributedSequentialSampler::next,
+          "Returns the next index if possible, or an empty optional if the sampler is exhausted for this epoch",
+          py::arg("batch_size"))
+      .def(
+          "save",
+          &DistributedSequentialSampler::save,
+          "Serializes the `RandomSampler` to the `archive`",
+          py::arg("archive"))
+      .def(
+          "load",
+          &DistributedSequentialSampler::load,
+          "Deserializes the `Sampler` from the `archive`",
+          py::arg("archive"))
+      .def(
+          "index",
+          &DistributedSequentialSampler::index,
+          "Returns the current index of the `DistributedSequentialSampler`");
+
+  /// RandomSampler
+  py::class_<RandomSampler>(data, "RandomSampler", distributed_sampler)
+      .def(
+          py::init<int64_t>(),
+          "Create and return a new `RandomSampler` instance",
+          py::arg("size"))
+      .def(
+          "reset",
+          &RandomSampler::reset,
+          "Resets the `Sampler`'s internal state. Typically called before a new epoch. Optionally, accepts a new size when reseting the sampler",
+          py::arg("new_size"))
+      .def(
+          "next",
+          &RandomSampler::next,
+          "Returns the next index if possible, or an empty optional if the sampler is exhausted for this epoch",
+          py::arg("batch_size"))
+      .def(
+          "save",
+          &RandomSampler::save,
+          "Serializes the `RandomSampler` to the `archive`",
+          py::arg("archive"))
+      .def(
+          "load",
+          &RandomSampler::load,
+          "Deserializes the `Sampler` from the `archive`",
+          py::arg("archive"))
+      .def(
+          "index",
+          &RandomSampler::index,
+          "Returns the current index of the `RandomSampler`");
+
+  /// SequentialSampler
+  py::class_<SequentialSampler>(data, "SequentialSampler", distributed_sampler)
```

Comment:
Same for those. They're already available in Python

Reply:
Same as before. We need a c++ sampler for a c++ dataset object